device/cuda/src/clusterization/clusterization_algorithm.cu

```diff
@@ -344,14 +373,23 @@ clusterization_algorithm::output_type clusterization_algorithm::operator()(
     CUDA_ERROR_CHECK(
         cudaMemset(num_measurements_device.get(), 0, sizeof(unsigned int)));
 
+    const unsigned short max_cells_per_partition =
+        m_target_cells_per_partition *
+        (MAX_CELLS_PER_THREAD / (float)TARGET_CELLS_PER_THREAD);
```

Comment:
This is the only thing I don't like. Floating point numbers are evil. Would suggest the following:

```suggestion
    const unsigned short max_cells_per_partition = (m_target_cells_per_partition * MAX_CELLS_PER_THREAD + TARGET_CELLS_PER_THREAD - 1) / TARGET_CELLS_PER_THREAD);
```